### ISSUE TITLE
Review files pointing to non-existing submission file may break upgrade to 3.3

### ISSUE THREAD

**Issue by Vitaliy-1** [pkp/pkp-lib#6743]
**Describe the bug**
Likely related forum topic: https://forum.pkp.sfu.ca/t/error-during-upgrade-to-ojs-3-3-0-3/65762

Consider entries in a review_files table like:
```
review_id	| file_id	
1	          67	
1	          68
...	
16              654
16              655
16              656
16              657 
```
If assume that submission files with id like 67 and 68 or 654-657 are missing, meaning multiple submission files for a given `review_id`, it will lead to a fatal error during upgrade:
```
ERROR: Upgrade failed: DB: SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`ojs-dev2`.`#sql-5ab_253`, CONSTRAINT `review_files_submission_file_id_foreign` FOREIGN KEY (`submission_file_id`) REFERENCES `submission_files` (`submission_file_id`)) (SQL: alter table `review_files` add constraint `review_files_submission_file_id_foreign` foreign key (`submission_file_id`) references `submission_files` (`submission_file_id`))
```
The reason behind this is that pluck method of the database querybuilder, in particular, used [here](https://github.com/pkp/pkp-lib/blob/00907578fa6f18e31a941de25e5842811fc0a471/classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php#L531), returns values (`submission_id`'s) only one per key (`review_id`). In the case of duplicates, it simply overrides the last value for a given key with a new one, it's described here: https://laravel.com/docs/8.x/collections#method-pluck.
In the example above it returns something like:
```
array 
(
  [1] => 68,
  [16] => 657
)
``` 
And it leads to submission_file_id values in `review_files` table that don't exist in `submission_files`.

**To Reproduce**
Try to upgrade a database with the data corrupted in the way described

**What application are you using?**
 reported for upgrade attempt from  OJS 3.2.1-3 to 3.3.0-3

**Additional information**
A part of https://github.com/pkp/pkp-lib/issues/6615
Special cases that arise from this commit: https://github.com/pkp/pkp-lib/commit/dde63cd8c6d4aad84c6d37bce0826a41141486df

**Comment by Vitaliy-1** [comment]
PRs
pkp-lib master: https://github.com/pkp/pkp-lib/pull/6744
pkp-lib stable-3_3_0: https://github.com/pkp/pkp-lib/pull/6745
tests (on master branch): https://github.com/pkp/ojs/pull/3039

**Comment by asmecher** [comment]
This looks good to me, @Vitaliy-1, and hopefully the user can confirm that it fixes the issue before we release 3.3.0-4. Please go ahead and merge.

**Comment by eirikhanssen** [comment]
I can confirm I had this same error when upgrading our largest OJS database: 
ERROR: Upgrade failed: DB: PDOException: SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (journals_stage.#sql-500_219b, CONSTRAINT review_files_submission_file_id_foreign FOREIGN KEY (submission_file_id) REFERENCES submission_files (submission_file_id)) in /var/www/vhosts/journals-stage.oslomet.no-ojs-3.3.0-3/lib/pkp/lib/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:115

And after applying https://github.com/pkp/pkp-lib/commit/6dd9e2a9dcc4307065c597fd96db66d56ee50578 I was able to complete the upgrade from OJS 3.2.1-4 to OJS 3.3.0-3.
